packages/cli/src/commands/optimize/optimize.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @flow
+ */
+
+const fs = require('fs').promises;
+const path = require('path');
+import sharp from 'sharp';
+import type {ConfigT} from 'types';
+import optimizeCommandLineArgs from './optimizeCommandLineArgs';
+import {logger} from '@react-native-community/cli-tools';
+
+function isImage(filePath: string) {
+  const regex = /\.(png|jpg|jpeg)/;
+  return filePath.toLowerCase().match(regex);
+}
+
+async function optimize(args: Array<string>, ctx: ConfigT) {
+  logger.log('Optimizing assets...');
+  const rootPathForImages = path.join(ctx.root, 'src');
+  const images = await getAllImageAssets(rootPathForImages);
+  for (const image of images) {
+    const buffer = await sharp(image).toBuffer();
+    const metadata = await sharp(buffer).metadata();
+    const newName = createOutputFilename(image);
+
+    const {mTimeMS: lastModified, ino: inode} = fs.statSync(image);
+    if (metadata.format === 'jpeg') {
+      sharp(image)
+        .jpeg({quality: 75})
+        .toFile(newName);
+    } else {
+      const outputFile = await sharp(image)
+        .png({quality: 75})
+        .toFile(newName);
+    }
+  }
+}
+
+/**
+ * Get all image assets in react native project
+ */
+const getAllImageAssets = async (dir, fileList = []) => {
```

Comment:
We have a utility for [walking directories](https://github.com/react-native-community/cli/blob/master/packages/cli/src/tools/walk.js), can you use that instead?